src/main/java/frc/robot/RobotContainer.java

```diff
@@ -1,145 +1,176 @@
+    SendableChooser<String> m_driverChooser = new SendableChooser<>();
+
+    // m_controllerType 0 -> Unrecognized
+    // m_controllerType 1 -> Xbox Controller
+    // m_controllerType 2 -> Playstation Controller
+    // m_controllerType 3 -> Flight Joystick
+    int m_controllerType;
+
+    /**
+     * The container for the robot. Contains subsystems, OI devices, and commands.
+     */
+    private RobotContainer() {
+        configureButtonBindings();
+
+        int pairButton;
+        String hidType = m_driverHID.getHID().getName();
+        if (hidType.equals("")) { // Xbox Controller | Name Unknown
+            m_controllerType = 1;
+            pairButton = 7;
+        } else if (hidType.equals("Wireless Controller")) { // PS5 | Is still called "Wireless Controller" if plugged in with a wire.
+            m_controllerType = 2;
+            pairButton = 7;
+        } else {
+            m_controllerType = 0;
+            pairButton = 7;
+        }
+
+
+        // ==================OPERATOR CONTROLS======================================
+
+        // Create Triggers here | Triggers should be named t_CommandName
+        Trigger leftTrigger = m_operatorHID.axisGreaterThan(3, Constants.DEADZONE);
+        Trigger rightTrigger = m_operatorHID.axisGreaterThan(4, Constants.DEADZONE);
+
+        Trigger operatorDriveTrigger = m_operatorHID.axisGreaterThan(1, Constants.DEADZONE);
+        operatorDriveTrigger.onTrue(m_operatorDrive);
+
+        Trigger operatorSpeedUp = m_operatorHID.button(2);
+        Trigger operatorSpeedDown = m_operatorHID.button(3);
+        operatorSpeedUp.onTrue(new setSpeedCommand(true, m_tankDriveSubsystem));
+        operatorSpeedDown.onTrue(new setSpeedCommand(false, m_tankDriveSubsystem));
+
+        Trigger intakeMoveUp = m_operatorHID.axisGreaterThan(1, Constants.DEADZONE);
+        Trigger intakeMoveDown = m_operatorHID.axisLessThan(1, -Constants.DEADZONE);
+        intakeMoveUp.whileTrue(new InstantCommand(() -> m_intake.setRaiseIntakeSpeed(0.1), m_intake));
+        intakeMoveDown.whileTrue(new InstantCommand(() -> m_intake.setRaiseIntakeSpeed(-0.1), m_intake));
+        intakeMoveUp.or(intakeMoveDown).onFalse(new InstantCommand(() -> m_intake.setRaiseIntakeSpeed(0), m_intake));
+
+        // ================DRIVER CONTROLS==========================================
+        // create commands
+        // 5 = left bumper
+        // 6 = right bumper
+
+        // Intake is toggled when left bumper is pressed
+        Trigger flipTrigger = m_driverHID.button(5);
+        flipTrigger.onTrue(m_intake.new FlipIntake(m_intake));
+
+        // Intake runs when right trigger is pressed
+        BooleanSupplier m_turnOnIntake = m_driverHID.axisGreaterThan(3, Constants.DEADZONE);
+        IntakeCommand m_IntakeCommand = new IntakeCommand(m_intake, m_turnOnIntake);
+        Trigger runIntakeTrigger = m_driverHID.button(3);
+        runIntakeTrigger.onTrue(m_IntakeCommand);
```

Comment:
I'm not sure that this code makes sense to me.
IntakeCommand's constructor is taking right trigger in as a boolean supplier.
When that trigger is true we run the intake. When it's false we don't. Good.

But then we attach button 3 to start the command? What's this doing?

Reply:
Button 3 shouldnt even be used i dont think

Reply:
The right bumper (which is 5) is suppose to reverse the intake & midtake, the right trigger turns the intake on and off.

Reply:
Think this should look like this:
`BooleanSupplier m_turnOnIntake = m_driverHID.button(5);
        IntakeCommand m_IntakeCommand = new IntakeCommand(m_intake, m_turnOnIntake);
        Trigger runIntakeTrigger = m_driverHID.axisGreaterThan(3, Constants.DEADZONE);
        runIntakeTrigger.onTrue(m_IntakeCommand);`

Reply:
At first glance that looks sensible.

If you have a controller handy and some time consider giving it a try?
Some things to remember:
- `execute` will keep running as long as the command is live. Comes through at 20ms period (I think)
- Unless `isFinished()` is implemented the execute command will keep running

I'm not trying to guide you to a specific bug with these points, just re-iterating something that @crazininja / @AkshayKuchibhatla  identified as a bug elsewhere.

Reply:
Speaking of controllers, I would like to add a remidner that we should be testing with two controllers because of the operator and the driver controls being separate.